### PR TITLE
Added user endpoint

### PR DESCRIPTION
**Pull request by stamlercas**
Added user endpoint to get information on user's overview, comments, upvotes, downvotes, etc.

### CODE REVIEW

File: .jshintrc [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "camelcase": true,
   "curly": true,
   "eqeqeq": true,
+  "esversion": 6,
```
Comment (sahilm) [inline .jshintrc]: Please remove this for now. Let's talk about moderninzing the library in your pull request.

File: reddit.js [hunk 2]
```diff
@@ -120,6 +120,13 @@
     });
   };
 
+  reddit.user = function (username, where = "") {
```
Comment (sahilm) [inline reddit.js]: Please change this to `es5` compatible code by removing the default parameter. This is until we modernize the lib.

**Comment by sahilm** [comment]
Thanks for the pull request. I've left some feedback for you.

Also I would be interested in knowing how you're using this library. A lot has changed in the JS world since I wrote this lib. I guess it's time I rewrite it in modern JS. What do you think?

File: reddit.js (hunk 2 above)
Comment (sahilm) [inline reddit.js]: There's an `aboutUser` resource [defined](https://github.com/sahilm/reddit.js/blob/master/reddit.js#L117-L121). I think that should be replaced with this new resource.

**Comment by stamlercas** [comment]
I agree with your feedback and I will make changes. Rewriting it might not be a bad idea. I was using this library to learn some JS frameworks.

**Comment by sahilm** [comment]
Thanks.

**Pull request merged**